twitter_api/main.py

```diff
@@ -1,28 +1,278 @@
 import sqlite3
-
-from flask import Flask
+import json
+from flask import Flask, Response, request, abort
 from flask import g
 
-app = Flask(__name__)
+# extra modules we added
+from hashlib import md5
+import random
+from .utils import JSON_MIME_TYPE, auth_only, json_only
+import string
+from datetime import datetime
 
+app = Flask(__name__)
+#api = Api(app)
 
 def connect_db(db_name):
     return sqlite3.connect(db_name)
 
-
 @app.before_request
 def before_request():
     g.db = connect_db(app.config['DATABASE'])
+    users = g.db.execute('SELECT access_token FROM auth')
+    
+# implement your views here
+@app.route('/login', methods = ['POST'])
+@json_only
+def login():
+    data = request.get_json()
+    try:
+        post_username = data['username']
+        post_password = data['password']
+    except KeyError:
+        abort(400) # Bad request
+    
+    sql_command = 'SELECT username, password, id FROM user WHERE username = ?'
+    
+    found_user_cursor = g.db.execute(sql_command, [post_username])
+    found_user = found_user_cursor.fetchone()
+    try:
+        username, password, id = found_user
+    except TypeError: # if username did find a match in db
+        abort(404) # Not found
+        
+    post_password = post_password.encode('utf-8') # needs to be in bytes
+    post_password = md5(post_password).hexdigest()
+    
+    if password != post_password:
+        abort(401) # Unauthorized
+        
+    # generate access token
+    char_list = string.ascii_lowercase + string.ascii_uppercase
+    access_token = ''.join(random.choice(char_list) for _ in range(10))
+    
+    sql_command = 'INSERT INTO auth (user_id, access_token) VALUES (?, ?)'
+    g.db.execute(sql_command, [id, access_token])
+    g.db.commit()
+    
+    return_dict = dict(access_token = access_token)
+    response = Response(
+        json.dumps(return_dict), status=201, content_type=JSON_MIME_TYPE)
+    return response
+    
+@app.route('/logout', methods = ['POST'])
+@json_only
+@auth_only
+def logout():
 
+    post_request = request.get_json()
+    post_access_token = post_request['access_token']
```

Comment:
What happens if the access_token is not sent? You'd fail with a KeyError. Check it and raise a 400 if it's not present.